src/utils.h

```diff
@@ -45,30 +45,8 @@
 
 #include <boost/uuid/detail/sha1.hpp>
 
-/**
- * @name Default values
- */
-/*@{*/
-
 #define TIMERS_ON   1
 
-#if DEBUG_MODE
-#define DEBUG_ON    1
-#define VERIFY_ON   1
-#define TRACING_ON  1
-#elif NO_ASSERTS
-#define DEBUG_ON    0
-#define VERIFY_ON   0
-#define TRACING_ON  0
-//#define NO_LOGGING  1
-#else
-#define DEBUG_ON    0
-#define VERIFY_ON   0
-#define TRACING_ON  1           // use --trace X to enable
-#endif
-
-/*@}*/
-
```

Comment:
I couldn't explain why these defines are set in `src/utils.h` as opposed to `src/system.hh.in`—except maybe for pre-compilation reasons—and thought it would be a good clean-up to keep all ledger related defines used by several source files in system.hh.in.

Can you think of any issues with this changes?

Reply:
They aren't in system.hh because they are system-level defines, they are specific to Ledger. System.hh really only exists to collect all of the non-Ledger code we want to import into a single PCH.

Reply:
I'll figure something out and come back with another proposal, if that's alright with you?

Reply:
@jwiegley What do you think about introducing `ledger.hh` (generated by CMake from `ledger.hh.in` similar to `system.hh.in`) that provides the version information as other application specific defines for the rest of the source (see 1fe7d71cf130074f2b345ebe70f9e76e8899b072)?

Reply:
Yes, that sounds like a good approach to me.

Reply:
`system.hh` really exists for only one purpose: to create the PCH. So we should avoid putting anything meaningful in there.

Reply:
I went ahead an [`DEBUG_MODE` and `NO_ASSERTS`](https://github.com/ledger/ledger/pull/2238/files#diff-bff4a86f314c161c10f267650141495d4921fd839bc8bfb1583d2a2b53a4585bR67-R68) should moved into [`ledger.hh.in`](https://github.com/ledger/ledger/pull/2238/files#diff-050964adfe522ee4701297304e4e126c6eef2bf6787cee3fea45dfb44d24b34c), @jwiegley, and took the liberty to re-order the system.hh includes to something that felt a bit more useful to me.

Please let me know if there is anything that you'd like to see changed before approving this PR, so it can be merged.